src/sampling.jl

```diff
@@ -316,12 +343,28 @@ function sample!(a::AbstractArray, x::AbstractArray; replace::Bool=true, ordered
     return x
 end
 
+
+"""
+    sample(a, [wv::WeightVec], n::Integer; replace=true, ordered=false)
+
+Select a random, optionally weighted sample of size `n` from an array `a`
+using a polyalgorithm. `replace` dictates whether sampling is performed
+with replacement and `order` dictates whether an ordered sample, also called
+a sequential sample, should be taken.
+"""
 function sample{T}(a::AbstractArray{T}, n::Integer; replace::Bool=true, ordered::Bool=false)
-    sample!(a, Array(T, n); replace=replace, ordered=ordered)
+    sample!(a, Array{T}(n); replace=replace, ordered=ordered)
 end
 
+
+"""
+    sample(a, [wv::WeightVec], dims::Dims; replace=true, ordered=false)
+
+Select a random, optionally weighted sample from an array `a` specifying
```

Comment:
The details provided for `sample!` should be repeated here. I must say I'm still not sure what this function does (but I don't know what polyalgorithm means): does it just draw random elements like the `n` variant, and returns them in a different form? or does it draw random slices? something else?